pkg/shares/parse_compact_shares.go

```diff
@@ -73,7 +63,12 @@ func parseRawData(rawData []byte) (units [][]byte, err error) {
 // not contain the namespace ID, info byte, sequence length, or reserved bytes.
 func extractRawData(shares []Share) (rawData []byte, err error) {
 	for i := 0; i < len(shares); i++ {
-		raw, err := shares[i].RawData()
+		var raw []byte
+		if i == 0 {
+			raw, err = shares[i].RawDataUsingReserved()
```

Comment:
[question] the reserved bytes exist for all compact shares. Why does the first share in `shares` uses `.RawDataUsingReserved()` while subsequent shares use `.RawData()`?

I think this function makes assumptions on the shares that are passed as a parameter, potentially that it is only called with share sequences (?) If so, can we document via godoc or better yet enforce via the type of the function argument?

Reply:
updated the doc string covering the comment.